### PR TITLE
Fix jit tls issue

### PR DESCRIPTION
**Pull request by yihuaxu**
According to the description of [Issue#1360](https://github.com/PaddlePaddle/models/issues/1360), the TLS of core_avx.so will occupy more size. Now we modified the source code of jit to decrease the TLS section size.

Before:
image

After:
image

### CODE REVIEW

File: paddle/fluid/operators/jit/helper.h [hunk 1]
```diff
@@ -175,13 +175,22 @@ typename KernelTuple::func_type GetDefaultBestFunc(
...
   KernelFuncs() = default;
   static KernelFuncs& Cache() {
-    static thread_local KernelFuncs<KernelTuple, PlaceType> g_func_cache;
-    return g_func_cache;
+    std::string key = typeid(KernelFuncs<KernelTuple, PlaceType>).name();
+    if (g_func_cache_map.find(key) != g_func_cache_map.end()) {
+      auto func_cache = g_func_cache_map.at(key);
+      return *(KernelFuncs<KernelTuple, PlaceType>*)(func_cache);
+    } else {
+      auto cache = new KernelFuncs<KernelTuple, PlaceType>;
```
Comment (luotao1) [inline paddle/fluid/operators/jit/helper.h]: new的部分需要做处理，或者用智能指针，不然会有内存泄漏。

Comment (yihuaxu) [reply paddle/fluid/operators/jit/helper.h]: Done!

**Comment by yihuaxu** [comment]
About the compiling failure of PR_CI_python35, it depends on GCC compiler(version:4.8.2 20140120).
As detail, please refer to the below link.
https://gcc.gnu.org/bugzilla/show_bug.cgi?id=59364

Log:
	benchmark.cc:(.text._ZN6paddle9operators3jit11KernelFuncsINS1_10GRUH1TupleIfEENS_8platform8CPUPlaceEE5CacheEv[_ZN6paddle9operators3jit11KernelFuncsINS1_10GRUH1TupleIfEENS_8platform8CPUPlaceEE5CacheEv]+0x102): undefined reference to `TLS init function for paddle::operators::jit::g_func_cache_map'

**Review by wanghaoshuang — APPROVED** [review]
LGTM

**Pull request merged**